### ISSUE TITLE
restart loop when accessing meetings from browser url 

### ISSUE THREAD

**Issue by maurerr** [IsmaelMartinez/teams-for-linux#646]
**Describe the bug**
I'm trying to join a teams meeting via a url like https://teams.microsoft.com/l/meetup-join/19%3ameeting_blahblah blahAw%40thread.v2/0?context=%7b%22Tid%22%3a%2211111111-1111-40bf-bc48-1111111120%22%2c%22Oid%22%3a%2294177b10-7452-1111-1111-1111111%22%7d in both firefox or chromium on ubuntu 22.04 x64

**To Reproduce**
1. Go to ' https://teams.microsoft.com/l/meetup-join/19%3ameeting_blahblah blahAw%40thread.v2/0?context=%7b%22Tid%22%3a%2211111111-1111-40bf-bc48-1111111120%22%2c%22Oid%22%3a%2294177b10-7452-1111-1111-1111111%22%7d'
2. Click on 'Open you teams app' or open link via system handler
3. Teams-for-linux snap starts to restart
4.  need to issue "killall teams-for-linux" to recover my system.

**Expected behavior**
Ability to join a meeting

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04 x86-64
 - Installation package: store snap
 - Version : 1.0.45

**Comment by IsmaelMartinez** [comment]
Hi @maurerr , If you copy the meeting link into a chat (or channel) and try to join from there, does it work?

I think MS have changed their URL handling and is breaking this and the links (what is a pain).

**Comment by maurerr** [comment]
when pasting the link into a chat or channel it doesn't work:
```
We couldn't find any results for 'https://teams.microsoft.com/l/meetup-join/19%3ameeting_blahblah%40thread.v2/0?context={"Tid"%3a"blah'
Check for spelling or try searching for another term

**Comment by jijojosephk** [comment]
@maurerr have you copied the complete url? Send a message to yourself and then click on that link. Let us know what happens

**Comment by maurerr** [comment]
> @maurerr have you copied the complete url? Send a message to yourself and then click on that link. Let us know what happens

yes complete url - teams is maybe truncating it in the search field

**Comment by DavidRayner** [comment]
I am seeing similar behaviour. Recently it was still working for me if I clicked the "continue on this browser" button at the right time as it was looping. But as of now, when clicking the "continue on this browser button" the app just shows a blue spinner indefinitely.

**Comment by IsmaelMartinez** [comment]
by the way, did any of you try another package version? Like deb or similar. Just to make sure is not related to snap doing one of its nonsense issues. Thanks in advance

**Comment by maurerr** [comment]
on a test virtual desktop debian-11 system it seem fine with a deb installation (unable to reproduce) but I want to confirm on my main work system (original reproducer)

**Comment by VGalata** [comment]
I have the same issue since today (edge-version `1.0.46`): first everything worked just fine then Teams closed itself (this has been happening for a couple of weeks now without further issues) and afterwards I could not open any links anymore. I then downgraded to `1.0.45` and could open one link but not any others afterwards...

System: Ubuntu 22.04
Teams-for-Linux: `1.0.45` (and `1.0.46`) installed via `snap`

**Comment by maurerr** [comment]
> by the way, did any of you try another package version? Like deb or similar. Just to make sure is not related to snap doing one of its nonsense issues. Thanks in advance

on my main system (ubuntu 22) it reproduces with deb as well :(

**Comment by maurerr** [comment]
I'm closing this - feel free to reopen if you have a reproducible scenario

**Comment by VGalata** [comment]
Could we re-open the ticket please? I still have the issue with the links and I have only the `snap` package installed (i.e., no deb package).

**Comment by maurerr** [comment]
> Could we re-open the ticket please? I still have the issue with the links and I have only the `snap` package installed (i.e., no deb package).

you don't have the "reopen" button active ?

**Comment by VGalata** [comment]
> you don't have the "reopen" button active ?

No, I do not see any.

**Comment by maurerr** [comment]
> > you don't have the "reopen" button active ?
> 
> No, I do not see any.

then I'll reopen it and unsubscribe from notifications

**Comment by maurerr** [comment]
@VGalata have you tried reinstalling the snap ?

**Comment by VGalata** [comment]
> @VGalata have you tried reinstalling the snap ?

@maurerr: do you mean the whole `snap` package or just the `teams-for-linux` app? I tried the latter only but have not re-installed `snap`.

**Comment by IsmaelMartinez** [comment]
Hi @VGalata ,

maybe try checking the configuration of your xdg config.

Something like this:
 https://techcommunity.microsoft.com/t5/microsoft-teams-blog/microsoft-teams-is-now-available-on-linux/bc-p/1064014/highlight/true#M4458

But using `teams-for-linux.desktop` instead of `msteams.desktop`

It can be that you use to have the msteams app installed and those haven't changed. Hopefully it helps.

**Comment by VGalata** [comment]
@maurerr Thanks for the tip. I removed the app and installed it again in `snap`. I will need to see today if it helped. Usually, the first link works but not (all) the others I use later.

@IsmaelMartinez Thank you for the pointer! However, the only xdg settings I have are these ones and there is no `msteams.desktop`:
```bash
xdg-settings --list
Known properties:
  default-url-scheme-handler    Default handler for URL scheme
  default-web-browser           Default web browser

xdg-settings get default-web-browser
firefox.desktop

xdg-settings get default-url-scheme-handler
```

**Comment by IsmaelMartinez** [comment]
I suspect then that if you configure the msteams as per the instructions on that link (but changing the app name as I indicated) it will probably solve your problem. As per why it doesn't show and/or get configured? I have no idea.

**Comment by VGalata** [comment]
I did what was described in the link you mentioned and it did not help. I tried to open a link multiple times but only see the loading icon and nothing happens. Also, it would have been strange if it did help because I did not have this issue until the recent update and sometimes the app does open a link.

Just for reference:
In `/home/galatav/.config/autostart/teams-for-linux_teams-for-linux.desktop`
```
MimeType=x-scheme-handler/msteams;
```
In `~/.local/share/applications/mimeapps.list`
```
[Default Applications]
x-scheme-handler/msteams=teams-for-linux_teams-for-linux.desktop
```

In `~/.config/mimeapps.list`
```
x-scheme-handler/msteams=teams-for-linux_teams-for-linux.desktop
```

Then:
```bash
sudo update-mime-database /usr/share/mime
update-desktop-database ~/.local/share/applications/
```
